crates/node/src/lib.rs

```diff
@@ -886,6 +901,25 @@ impl Node {
         Ok(outcome)
     }
 
+    async fn reject_from_pool(
+        &mut self,
+        hash: calimero_primitives::hash::Hash,
+    ) -> eyre::Result<()> {
+        while let Some(transaction_pool::TransactionPoolEntry { transaction, .. }) =
+            self.tx_pool.remove(&self.last_tx)
```

Comment:
ah, we're gonna need to remove last_tx from `Node` and read it from `ContextMeta` now